lib/iris/experimental/ugrid/__init__.py

```diff
@@ -2746,6 +2747,239 @@ def face_node(self):
+        )
+
+        file_path = sample_data_path("mesh_C4_synthetic_float.nc")
+        with PARSE_UGRID_ON_LOAD.context():
+            cube_w_mesh = load_cube(file_path)
+
+    For example::
+
+        # Reconstruct a cube-with-mesh after subsetting it.
+
+        >>> print(cube_w_mesh.mesh.name())
+        Topology data of 2D unstructured mesh
+        >>> mesh_coord_names = [
+        ...     coord.name() for coord in cube_w_mesh.coords(mesh_coords=True)
+        ... ]
+        >>> print(f"MeshCoords: {mesh_coord_names}")
+        MeshCoords: ['latitude', 'longitude']
+
+        # Subsetting converts MeshCoords to AuxCoords.
+        >>> slices = [slice(None)] * cube_w_mesh.ndim
+        >>> slices[cube_w_mesh.mesh_dim()] = slice(-1)
+        >>> cube_sub = cube_w_mesh[tuple(slices)]
+        >>> print(cube_sub.mesh)
+        None
+        >>> orig_mesh_coords = [cube_sub.coord(c_name) for c_name in mesh_coord_names]
+        >>> for coord in orig_mesh_coords:
+        ...     print(f"{coord.name()}: {type(coord).__name__}")
+        latitude: AuxCoord
+        longitude: AuxCoord
+
+        >>> new_mesh = mesh_from_coords(orig_mesh_coords[0], orig_mesh_coords[1])
+        >>> new_mesh_coords = new_mesh.to_MeshCoords(location=cube_w_mesh.location)
+
+        # Replace the AuxCoords with MeshCoords.
+        >>> for ix in range(2):
+        ...     cube_sub.remove_coord(orig_mesh_coords[ix])
+        ...     cube_sub.add_aux_coord(new_mesh_coords[ix], cube_w_mesh.mesh_dim())
+
+        >>> print(cube_sub.mesh.name())
+        Topology data of 2D unstructured mesh
+        >>> for coord_name in mesh_coord_names:
+        ...     coord = cube_sub.coord(coord_name)
+        ...     print(f"{coord_name}: {type(coord).__name__}")
+        latitude: MeshCoord
+        longitude: MeshCoord
+
+    """
+    # Validate coord essentials.
+    coord_names = {coord_1: "coord_1", coord_2: "coord_2"}
+    for coord, name in coord_names.items():
+        if not isinstance(coord, Coord):
+            message = f"Expected a Coord for {name}, got: {type(coord)}"
+            raise ValueError(message)
+
+        if coord.ndim != 1:
+            message = f"Coordinates must have ndim == 1. f{name}.ndim == {coord.ndim} ."
+            raise ValueError(message)
+
+        invalid_bounds = False
+        sub_message = ""
+        if not coord.has_bounds():
+            invalid_bounds = True
+            sub_message = "None"
+        else:
+            bounds_shape = coord.core_bounds().shape
+            # 1D coord check guarantees bounds are ndim==2.
+            if bounds_shape[1] < 2:
+                invalid_bounds = 2
+                sub_message = f"shape {bounds_shape}. Must be (n, >=2)."
+        if invalid_bounds:
+            message = f"{name} has invalid bounds: " + sub_message
+            raise ValueError(message)
+
+    # Validate orthogonality.
+    coord_axes = {
+        coord: guess_coord_axis(coord) for coord in (coord_1, coord_2)
```

Comment:
As suggested above, this may make more sense as a list or tuple.

Reply:
https://github.com/SciTools/iris/pull/4326/files#r715711539